lib/transformers/tokend.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const http = require('http');
+const Immutable = require('immutable');
+
+/**
+ * Walk a properties object looking for transformable values
+ *
+ * @param {Object} properties  an object to search for transformable values
+ * @param {Array} keyPath  accumulated path of keys where transformable value was found
+ * @return {Array<Immutable.Map>} transformable data with injected key path
+ */
+function collectTransformables(properties, keyPath) {
+  let results = [];
+
+  // Ensure we're walking an Object. `== null` covers both null and undefined.
+  if (properties == null || Object.getPrototypeOf(properties) !== Object.prototype) { // eslint-disable-line eqeqeq
+    return results;
+  }
+
+  // If we're walking a $tokend object, pass it to the callback.
+  const keys = Object.keys(properties);
+
+  if (keys.length === 1 && keys[0] === '$tokend') {
```

Comment:
I'm just going to make a note that we'll definitely have to revisit this when we add more transformers.